### DISCUSSION TITLE
Searching a specific attribute value in an array

### DISCUSSION THREAD

**Discussion by NB2909** [schibsted/jslt#369] [Q&A] [answered]
Hi! 

This is my first question here so please bear with me! 
I only have a very basic understanding of JSLT and also not looking to very deep dive into it. I did not find anything that I could try myself in the basic tutorial nor the functions. 

I am trying to determine the value of my attribute based on if a specific value is present in one of the array values. 

Example: I am looking if the "test1234" attribute has the specific value **xyz**. If one of them has this value, the outcome should show a true. If not, it will be a false
{
    "attribute1": "This is a test",
    "array": [
        {
            "test1234": "**xyz**"
        },
        {
            "test1234": "abc"
        },
        {
            "test1234": "cde"
        }
    ]
} 

Expected outcome:

{
    "attribute1": "This is a test",
    "attribute2": **true** // or **false** if no **xyz** is found. Or could also be a string based on the transformation, I do not mind about that
}

Example 2 with xyz not being present:
{
    "attribute1": "This is a test",
    "array": [
        {
            "test1234": "abc"
        },
        {
            "test1234": "cde"
        }
    ]
} 

{
    "attribute1": "This is a test",
    "attribute2": **false** 
}

Very much appreciate your help! 

Thanks, 
Nico

**Comment by catull** [comment, marked as answer]
Here's a solution.

Input

```json
{
  "attribute1": "This is a test",
  "array": [
    { "test1234": "abc" },
    { "test1234": "cde" },
    { "test1234": "xyz" }
  ]
}
```

With this transformation:

```
{
  "attribute1": .attribute1,
  "attribute2": contains ("xyz", [ for (.array) .test1234 ])
}
```

The result is:

```json
{
  "attribute1" : "This is a test",
  "attribute2" : true
}
```

**Reply by NB2909** [reply]
Awesome! Thanks a lot for the quick reply! I did not think about switching the contain and for function in my tries.
